core/src/main/java/org/kiwix/kiwixmobile/core/main/TableDrawerAdapter.kt

```diff
@@ -0,0 +1,120 @@
+
+import android.graphics.Typeface
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import kotlinx.android.synthetic.main.section_list.titleText
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.base.adapter.BaseViewHolder
+import org.kiwix.kiwixmobile.core.extensions.ViewGroupExtensions.inflate
+import java.lang.IllegalStateException
+
+class TableDrawerAdapter constructor(private val listener: TableClickListener) :
+  Adapter<ViewHolder>() {
+
+  private var title: String = ""
+  private val sections: MutableList<DocumentSection> = mutableListOf()
+
+  fun setSections(sections: List<DocumentSection>) {
+    this.sections.clear()
+    this.sections.addAll(sections)
+  }
+
+  fun setTitle(title: String) {
+    this.title = title
+  }
+
+  override fun getItemViewType(position: Int): Int {
+    return if (position == 0) 0
+    else 1
+  }
+
+  override fun onCreateViewHolder(
+    parent: ViewGroup,
+    viewType: Int
+  ): ViewHolder {
+    val view = parent.inflate(R.layout.section_list, false)
+    return if (viewType == 0) {
+      HeaderTableDrawerViewHolder(view)
+    } else {
+      SectionTableDrawerViewHolder(view)
+    }
+  }
+
+  override fun getItemCount(): Int = sections.size + 1
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    when (holder) {
+      is HeaderTableDrawerViewHolder -> {
+        holder.bind(title)
+        holder.itemView.setOnClickListener(listener::onHeaderClick)
+      }
+      is SectionTableDrawerViewHolder -> {
+        holder.bind(sections[position - 1])
+        holder.itemView.setOnClickListener {
+          listener.onSectionClick(it, position - 1)
+        }
+      }
+      else -> {
+        throw IllegalStateException("Unknown ViewHolder $holder found")
+      }
+    }
+  }
+
+  interface TableClickListener {
+    fun onHeaderClick(view: View?)
+    fun onSectionClick(view: View?, position: Int)
+  }
+
+  class HeaderTableDrawerViewHolder(view: View) :
+    BaseViewHolder<String>(view) {
+
+    override fun bind(item: String) {
+      val context = itemView.context
+      titleText.typeface = Typeface.DEFAULT_BOLD
+      titleText.text = when {
+        item.isNotEmpty() -> item
+        context is WebViewProvider -> context.getCurrentWebView()?.title
+          ?: context.getString(R.string.no_section_info)
+        else -> context.getString(R.string.no_section_info)
+      }
+    }
+  }
+
+  class SectionTableDrawerViewHolder(view: View) :
+    BaseViewHolder<TableDrawerAdapter.DocumentSection>(view) {
+    override fun bind(
+      item: TableDrawerAdapter.DocumentSection
+    ) {
+      val context = itemView.context
+      val density = context.resources.displayMetrics.density
+      val padding =
+        ((item.level - 1) * context.resources.getDimension(R.dimen.title_text_padding) * density)
```

Comment:
you no longer need to multiply by density, `dp` means `density pixel`, when we retrieve it from resources it automatically gets scaled by the density

![image](https://user-images.githubusercontent.com/3358282/95311257-45d21300-0885-11eb-9c8f-2ed1132c0e39.png)